### ISSUE TITLE
make docker-compose-build fails with conflicting dependencies

### ISSUE THREAD

**Issue by jean-christophe-manciot** [ansible/awx#15597]
### Please confirm the following

- [X] I agree to follow this project's [code of conduct](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html).
- [X] I have checked the [current issues](https://github.com/ansible/awx/issues) for duplicates.
- [X] I understand that AWX is open source software provided for free and that I might not receive a timely response.
- [X] I am **NOT** reporting a (potential) security vulnerability. (These should be emailed to `security@ansible.com` instead.)

### Bug Summary

The conflict is caused by:
- The user requested sqlparse==0.4.4
- django 4.2.10 depends on sqlparse>=0.3.1
- django-ansible-base[jwt-consumer,rbac,resource-registry,rest-filters] 2024.10.22.0.dev2+gb082f48 depends on sqlparse>=0.5.0

### AWX version

24.6.0

### Select the relevant components

- [ ] UI
- [ ] UI (tech preview)
- [ ] API
- [ ] Docs
- [ ] Collection
- [ ] CLI
- [X] Other

### Installation method

docker development environment

### Modifications

no

### Ansible version

2.17.5

### Operating system

CentOS stream 9

### Steps to reproduce

with:

- awx_version=24.6.0
- awx_devel_container_version=release_4.6
- awx_receptor_version=devel or latest
```
git clone -b $awx_version https://github.com/ansible/awx.git git-awx && cd git-awx
git switch -c $awx_devel_container_version
export RECEPTOR_IMAGE=quay.io/ansible/receptor:${awx_receptor_version}
make docker-compose-build
```

### Expected results

no errors

### Actual results

```
...
    ansible-playbook -e ansible_python_interpreter=python3.11 tools/ansible/dockerfile.yml \
            -e dockerfile_name=Dockerfile.dev \
            -e build_dev=True \
            -e receptor_image=quay.io/ansible/receptor:devel
...
    140.0 ERROR: Cannot install -r /dev/stdin (line 130), django-ansible-base[jwt-consumer,rbac,resource-registry,rest-filters]==2024.10.22.0.dev2+gb082f48 and sqlparse==0.4.4 because these package versions have conflicting dependencies.
    140.0 ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/user_guide/#fixing-conflicting-dependencies
    140.0 WARNING: You are using pip version 21.2.4; however, version 24.2 is available.
    140.0 You should consider upgrading via the '/var/lib/awx/venv/awx/bin/python3.11 -m pip install --upgrade pip' command.
    140.0
    140.0 The conflict is caused by:
    140.0     The user requested sqlparse==0.4.4
    140.0     django 4.2.10 depends on sqlparse>=0.3.1
    140.0     django-ansible-base[jwt-consumer,rbac,resource-registry,rest-filters] 2024.10.22.0.dev2+gb082f48 depends on sqlparse>=0.5.0
    140.0
...
```

**Comment by jean-christophe-manciot** [comment]
To avoid those frequent dependencies issues, is it possible to replace:
```
$ export RECEPTOR_IMAGE=quay.io/ansible/receptor:${awx_receptor_version}
$ make docker-compose-build
```
with
```
docker pull ghcr.io/ansible/awx_devel:${awx_devel_container_version}
```
That sounds so much simpler.

**Comment by jean-christophe-manciot** [comment]
It turns out it was not a stupid idea as I have just realized it has already been discussed several months ago  [here](https://github.com/ansible/awx/issues/15082).

**Comment by caboteria** [comment]
If this is the same bug I tripped over, it's fixed by c85fa70745
